connector/src/main/java/org/geysermc/connector/utils/AdvancementsUtils.java

```diff
@@ -0,0 +1,163 @@
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.connector.utils;
+
+import com.github.steveice10.mc.protocol.data.MagicValues;
+import com.github.steveice10.mc.protocol.data.game.advancement.Advancement;
+import com.github.steveice10.mc.protocol.data.game.entity.type.EntityType;
+import com.github.steveice10.mc.protocol.data.game.statistic.*;
+import org.geysermc.common.window.SimpleFormWindow;
+import org.geysermc.common.window.button.FormButton;
+import org.geysermc.common.window.response.SimpleFormResponse;
+import org.geysermc.connector.network.session.GeyserSession;
+import org.geysermc.connector.network.translators.chat.MessageTranslator;
+import org.geysermc.connector.network.translators.item.ItemRegistry;
+import org.geysermc.connector.network.translators.java.JavaAdvancementsTranslator;
+import org.geysermc.connector.network.translators.world.block.BlockTranslator;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AdvancementsUtils {
+
+    // Used in UpstreamPacketHandler.java
+    public static final int ADVANCEMENTS_MENU_FORM_ID = 1341;
+    public static final int ADVANCEMENTS_LIST_FORM_ID = 1342;
+
+    /**
+     * Build a form for the given session with all advancement categories
+     *
+     * @param session The session to build the form for
+     */
+    public static SimpleFormWindow buildMenuForm(GeyserSession session) {
+        // Cache the language for cleaner access
+        String language = session.getClientData().getLanguageCode();
+        // Created menu window for advancement categories
+        SimpleFormWindow window = new SimpleFormWindow(LocaleUtils.getLocaleString("gui.advancements", language), "");
+
+        window.getButtons().add(new FormButton(LocaleUtils.getLocaleString("advancements.story.root.title", language) + " - " + LocaleUtils.getLocaleString("advancements.story.root.description", language)));
+        window.getButtons().add(new FormButton(LocaleUtils.getLocaleString("advancements.nether.root.title", language) + " - " + LocaleUtils.getLocaleString("advancements.nether.root.description", language)));
+        window.getButtons().add(new FormButton(LocaleUtils.getLocaleString("advancements.end.root.title", language) + " - " + LocaleUtils.getLocaleString("advancements.end.root.description", language)));
+        window.getButtons().add(new FormButton(LocaleUtils.getLocaleString("advancements.adventure.root.title", language) + " - " + LocaleUtils.getLocaleString("advancements.adventure.root.description", language)));
+        window.getButtons().add(new FormButton(LocaleUtils.getLocaleString("advancements.husbandry.root.title", language) + " - " + LocaleUtils.getLocaleString("advancements.husbandry.root.description", language)));
+
+        return window;
+    }
+
+    /**
+     * Handle the menu form response
+     *
+     * @param session  The session that sent the response
```

Comment:
Remove extra space in between `session` and `The`.

Reply:
Extra space isn't showing for me in intellij